src/firefly/java/edu/caltech/ipac/firefly/server/servlets/AnyFileDownload.java

```diff
@@ -165,7 +165,18 @@ private static void logActivity(File f) {
     }
 
     private static void trackProgress(HttpServletRequest req, BackgroundEnv.DownloadProgress progress) {
-        StringKey statusKey= new StringKey(req.getQueryString());
+        String queryString =req.getQueryString();
+
+        if (queryString==null){
+            try {
+                throw new Exception("Status can not be tracked because the queryString is empty");
+            } catch (Exception e) {
+                Logger.error("HttpServletRequest has a null query string, the status can not be tracked");
+
+
+            }
```

Comment:
Why throw an exception just to catch it?  Do you want to log it?  Do you want execution to stop?  Do you want the caller to handle the exception?  What do you want it to do?

Reply:
This is the part of the bug I found.  The query status is NULL, thus, the download progress run forever.  I fixed the status.  In order to guard it,  I added the exception to log the error.

Reply:
I agree with @loitly this code is confusing.  The code is working just like before. It is just logging a message.  Why not just do this?
```
if (queryString==null){
         Logger.error("HttpServletRequest has a null query string, the status can not be tracked");
}
```

Reply:
If it works before, I would change it.  I am OK to remove the exception, however, if the null string comes again, it may need to debug it again.

Reply:
I reverted this file the same as the one in dev.  The "UNKNOWN" status is assigned when the cache key (added in this function) does not match the key received from the client in the BackgroundEnv.